src/vizier/services/query_broker/script_runner/config_map_source.go

```diff
@@ -0,0 +1,124 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package scriptrunner
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"px.dev/pixie/src/shared/cvmsgspb"
+	"px.dev/pixie/src/utils"
+)
+
+func ConfigMapSource(client clientv1.ConfigMapInterface) Source {
+	return func(baseCtx context.Context, updatesCh chan *cvmsgspb.CronScriptUpdate) (map[string]*cvmsgspb.CronScript, func(), error) {
+		options := metav1.ListOptions{LabelSelector: "purpose=cron-script"}
+		watcher, err := client.Watch(baseCtx, options)
+		if err != nil {
+			return nil, nil, err
+		}
+		go configMapUpdater(watcher, updatesCh)
+		configmaps, err := client.List(baseCtx, options)
+		if err != nil {
+			watcher.Stop()
+			return nil, nil, err
+		}
+		scripts := map[string]*cvmsgspb.CronScript{}
+		for _, configmap := range configmaps.Items {
+			id, cronScript, err := configmapToCronScript(&configmap)
+			if err != nil {
+				logCronScriptParseError(err)
+				continue
+			}
+			scripts[id] = cronScript
+		}
+		return scripts, watcher.Stop, nil
+	}
+}
+
+func configMapUpdater(watcher watch.Interface, updatesCh chan *cvmsgspb.CronScriptUpdate) {
+	for event := range watcher.ResultChan() {
+		switch event.Type {
+		case watch.Modified, watch.Added:
+			configmap := event.Object.(*v1.ConfigMap)
+			id, script, err := configmapToCronScript(configmap)
+			if err != nil {
+				logCronScriptParseError(err)
+				continue
+			}
+			cronScriptUpdate := &cvmsgspb.CronScriptUpdate{
+				Msg: &cvmsgspb.CronScriptUpdate_UpsertReq{
+					UpsertReq: &cvmsgspb.RegisterOrUpdateCronScriptRequest{
+						Script: script,
+					},
+				},
+				RequestID: id,
+				Timestamp: time.Now().Unix(),
+			}
+			updatesCh <- cronScriptUpdate
+		case watch.Deleted:
+			configmap := event.Object.(*v1.ConfigMap)
+			id, script, err := configmapToCronScript(configmap)
+			if err != nil {
+				logCronScriptParseError(err)
+				continue
+			}
+			cronScriptUpdate := &cvmsgspb.CronScriptUpdate{
+				Msg: &cvmsgspb.CronScriptUpdate_DeleteReq{
+					DeleteReq: &cvmsgspb.DeleteCronScriptRequest{
+						ScriptID: script.ID,
+					},
+				},
+				RequestID: id,
+				Timestamp: time.Now().Unix(),
+			}
+			updatesCh <- cronScriptUpdate
+		}
+	}
+}
+
+func logCronScriptParseError(err error) {
+	log.WithError(err).Error("Failed to parse cron.yaml from configmap cron script")
+}
+
+func configmapToCronScript(configmap *v1.ConfigMap) (string, *cvmsgspb.CronScript, error) {
+	id := string(configmap.UID)
+	cronScript := &cvmsgspb.CronScript{
+		ID:      utils.ProtoFromUUIDStrOrNil(id),
+		Script:  configmap.Data["script.pxl"],
```

Comment:
What does an example configmap look like?

Reply:
Do you have a preferred location to document this?

Reply:
We should probably add this new mechanism + an example configmap to our docs. Probably on this page: https://docs.px.dev/tutorials/integrations/otel/ 

But for now, just for code review purposes you can just send an example as a comment here.

Reply:
I also added a description to the doc string for the `ConfigMapSource` function.

```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: wavefront-span-script
  namespace: pl
  labels:
    purpose: cron-script
data:
  script.pxl: |
    px.export(...)
  configs.yaml: |
    otelEndpointConfig:
      insecure: true
      url: someurl:999
  cron.yaml: |
    frequency_s: 15
```